src/app/tests/suites/commands/interaction_model/InteractionModel.cpp

```diff
@@ -449,35 +449,129 @@ void InteractionModelReports::CleanupReadClient(ReadClient * aReadClient)
         mReadClients.end());
 }
 
-CHIP_ERROR InteractionModelReports::ReadAll(DeviceProxy * device, std::vector<EndpointId> endpointIds,
-                                            const Optional<bool> & fabricFiltered)
+CHIP_ERROR InteractionModelReports::ReportAll(chip::DeviceProxy * device, std::vector<chip::EndpointId> endpointIds,
+                                              std::vector<chip::ClusterId> clusterIds, std::vector<chip::AttributeId> attributeIds,
+                                              std::vector<chip::EventId> eventIds,
+                                              chip::app::ReadClient::InteractionType interactionType, uint16_t minInterval,
+                                              uint16_t maxInterval, const chip::Optional<bool> & fabricFiltered,
+                                              const chip::Optional<std::vector<chip::DataVersion>> & dataVersions,
+                                              const chip::Optional<chip::EventNumber> & eventNumber,
+                                              const chip::Optional<bool> & keepSubscriptions)
 {
+    const size_t endpointCount  = endpointIds.size();
+    const size_t clusterCount   = clusterIds.size();
+    const size_t attributeCount = attributeIds.size();
+    const size_t eventCount     = eventIds.size();
+
+    // TODO Add data version supports
+    // TODO Add isUrgents supports
+
+    VerifyOrReturnError(endpointCount > 0 && endpointCount <= kMaxAllowedPaths, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(clusterCount > 0 && clusterCount <= kMaxAllowedPaths, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(attributeCount > 0 && attributeCount <= kMaxAllowedPaths, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(eventCount > 0 && eventCount <= kMaxAllowedPaths, CHIP_ERROR_INVALID_ARGUMENT);
+
+    const bool hasSameIdsCount = (clusterCount == (attributeCount + eventCount)) && (clusterCount == endpointCount);
+    if (!hasSameIdsCount)
+    {
+        ChipLogError(chipTool,
+                     "\nCommand targetting a combination of attribute and event paths needs to have has many clusters and "
+                     "endpoints than the number of attribute and events combined.\n"
```

Comment:
"... needs to have the same number of clusters and endpoints specified as there are attribute and event ids combined".